File: algo/lc.88.cc
class Solution {
public:
    void merge(vector<int>& nums1, int m, vector<int>& nums2, int n) {
        int cur1 = m-1, cur2 = n-1, curh = m+n-1;
        while (cur1 >= 0 && cur2 >= 0) {
            if (nums1[cur1] > nums2[cur2]) {
                nums1[curh] = nums1[cur1];
                curh--;
                cur1--;
            } else { // <=
                nums1[curh] = nums2[cur2];
                curh--;
                cur2--;
            }
        }
        while (cur2 >= 0) {
            nums1[curh] = nums2[cur2];
            curh--;
            cur2--;
        }
    }
};

File: algo/lc.88.py
class Solution:
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        """
        Do not return anything, modify nums1 in-place instead.
        """
        if not nums1 and not nums2:
            return []
        elif not nums1 or not nums2:
            return nums2 if not nums2 else nums1
        # else
        for cur2 in range(n):
            # where to insert nums2[cur2]
            cur1 = 0
            while nums2[cur2] >= nums1[cur1] and cur1 < m + cur2:
                cur1 = cur1 + 1
            nums1.insert(cur1, nums2[cur2])
            nums1.pop(-1)
